packages/debug/src/browser/debug-session-manager.ts

```diff
@@ -214,12 +231,13 @@ export class DebugSessionManager {
             this.updateCurrentSession(session);
         });
         session.onDidChangeBreakpoints(uri => this.fireDidChangeBreakpoints({ session, uri }));
-        session.on('terminated', event => {
+        session.on('terminated', async event => {
             const restart = event.body && event.body.restart;
             if (restart) {
                 this.doRestart(session, restart);
```

Comment:
What if someone triggers restart from UI?

VS Code seems to run `postDebugTask` then: https://github.com/TypeFox/vscode/blob/70b8db24a37fafc77247de7f7cb5bb0195120ed0/src/vs/workbench/contrib/debug/browser/debugService.ts#L553-L559

but does not run on auto restart, i.e. from this if